contracts/PushComm/PushCommV3.sol

```diff
@@ -471,4 +474,67 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3 {
         userToChannelNotifs[msg.sender][_channel] = notifSetting;
         emit UserNotifcationSettingsAdded(_channel, msg.sender, _notifID, notifSetting);
     }
+
+    ///@notice Wallet PGP attach code starts here
+
+    function setFeeAmount(uint256 _feeAmount) external onlyPushChannelAdmin {
+        FEE_AMOUNT = _feeAmount;
+    }
+    /* *****************************
+
+         USER PGP Registry Functions
+
+    ***************************** */
+    function registerUserPGP(bytes calldata _data, string calldata _pgp, bool _isNFT) external {
+        uint256 fee = FEE_AMOUNT;
+        PROTOCOL_POOL_FEE += fee;
+        IERC20(PUSH_TOKEN_ADDRESS).safeTransferFrom(msg.sender, address(this), fee);
+        bytes32 hash = keccak256(_data);
```

Comment:
Rename args & Variables to enhance readability:

1. Change args _date to _caipData
2. Change local variable hash to caipHash